compiler/frontend.java/src/org/jetbrains/kotlin/resolve/jvm/diagnostics/DefaultErrorMessagesJvm.java

```diff
@@ -211,6 +211,8 @@
 
         MAP.put(TYPEOF_SUSPEND_TYPE, "Suspend functional types are not supported in typeOf");
         MAP.put(TYPEOF_NON_REIFIED_TYPE_PARAMETER_WITH_RECURSIVE_BOUND, "Non-reified type parameters with recursive bounds are not supported yet: {0}", STRING);
+
+        MAP.put(ILLEGAL_ACCESS, "tried to access class {0} from class {1}", STRING, STRING);
```

Comment:
This error diagnostics captures a rather specific situation: an implicit cast to non-accessible class. It's better to tell it straight in the error message: `"Implicit cast to non-accessible class ''{0}''"`.

Reply:
Also, rename the diagnostic factory itself accordingly.